File: Java/551. Student Attendance Record I.java
/**
 * https://leetcode.com/problems/student-attendance-record-i/
 */

class Solution {
    // O(n) time complexity
    // O(1) space complexity
    public boolean checkRecord(String s) {
        if (s == null || s.length() == 0) return true;
        int absent = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == 'A')
                absent++;
            if (i <= s.length() - 3
                    && s.charAt(i) == 'L' && s.charAt(i + 1) == 'L' && s.charAt(i + 2) == 'L')
                return false;
        }
        return absent < 2;
    }
}

File: Kotlin/551. Student Attendance Record I.kt
/**
 * https://leetcode.com/problems/student-attendance-record-i/
 */

class Solution {
    // O(n) time complexity
    // O(1) space complexity
    fun checkRecord(s: String): Boolean {
        if (s.isEmpty()) return true
        var absent = 0
        for (i in s.indices) {
            if (s[i] == 'A')
                absent++
            if (i <= s.length - 3 && s[i] == 'L' && s[i + 1] == 'L' && s[i + 2] == 'L')
                return false
        }
        return absent < 2
    }
}